### PR TITLE
Reorganize Terraform Cloud Sidebar

### PR DESCRIPTION
**Pull request by laurapacilio**
### What
This pull request is part 1 of an effort to revamp the TFC/TFE sidebars in anticipation of versioned documentation for Terraform Enterprise. I will open a separate PR for additional changes to the TFE sidebar.

Specifically, this PR:

- **Removes the / from the title.** It used to be "Terraform Cloud / Enterprise". The slash is a bit disorienting (and a bit ugly), and there's no reason to leave it that way when "and" continues to fit on a single line.
- **Reorders the Terraform Cloud Sidebar**: The current sidebar is not organized in a way to best serve users. The new sidebar moves API up to the top, so that users who come to this documentation set for API commands can easily navigate to the pages they need. This is aligned with the [Consul documentation](https://www.consul.io/docs/download-tools). Then, the rest of the sidebar entries are organized in order of importance. Core concepts and explanations of key workflows should come first, as users who are new to Terraform Cloud may use the sidebar to learn about our product. 
- **Renames pages in the sidebar:** Some of the original section names may not make it clear to new users what is on the page. Renaming these pages so that they are clearer can help users find the content they need, especially since our search engine indexes on titles. For example, the "Overview" page inside "Terraform Runs and Remote Operations" is really about "Remote Operations" and how Terraform queues and manages runs in remote execution mode. Before, users may have skipped  the "overview" page, as overview pages are sometimes simply summaries of the pages underneath. Now, they will know that this, specifically is where they can find information about remote ops, and this page will come up in search if users search for those keywords. Other changes include adding context to "Sentinel"--> "Policy as Code" so that users can understand its value and why they may want to click it. Also changing "VCS Integration" to "Connecting to VCS", which better describes the page content and reduces confusion about how that is the same/different from the other "Integrations" section.
- **Combines Features and Plans pages:** Over a period from Feb 1, 2022 - April 29, 2022, the features and plans pages only got 10.9k and 11.4k views, respectively. Contrast this with our most popular pages like Variables and "Intro to Terraform" which got 298k and 138k views, respectively. We are using valuable real estate at the top of the navigation to showcase this content, which is not very popular. By combining these pages together, we help users get to the information they actually need more quickly. And, a shorter sidebar is just less visually overwhelming :-). **Redirect has been added**.
- **Moves Best Practices up a level:** Best practices is the only subheading under "Guides". Whenever there is a single item underneath a top-level heading and we do not plan to add more items, this is wasted real estate--> adding more levels to the sidebar than is necessary. Also, "Guides" is misleading. These are not "how-to" articles. Rather, they are more like white papers and traditional "best practices" documentation that are written at a high level and meant for leads and decision makers. By moving this up a level, we have reduced sidebar depth (a win) and made it clearer what type of information is actually included in this section. **Redirect has been added**.
- **Fixes a few style nits** in the content and reworks the overview page so that it's easier to read and the Learn callouts appear sooner on the page. 
- **Removes the large header with the single "Terraform Enterprise" Deployment and Operation** and includes it in the sidebar in the same pattern as "Cloud Agents". This make logical sense --> it's related to the core TFC documentation set the way that cloud agents are, but it links users to a completely separate sidebar.  This helps the sidebar look less busy (ANOTHER bolded heading with "Other Docs" felt a bit visually overwhelming) and hopefully(?) helps to show that the doc sets are related.
- **Renames Operation and Deployment to "Terraform Enterprise Admin** Not only is this shorter, so it allows us to put "Terraform Enterprise" in front of it (which is an important distinction for users to understand that this sidebar is Enterprise-specific), but it also (I think) more accurately captures what is included in that documentation. This documentation is for administrating the TFE app. It's not for end users, it's for system admins and other operators who are in charge of maintaining the instance. This will go away when we do versioned docs, but I think it's a small change that is helpful until that occurs.

#### Before

Screen Shot 2022-05-11 at 3 05 50 PM

Screen Shot 2022-05-11 at 4 08 45 PM

--------------------

#### After
Screen Shot 2022-05-16 at 5 29 49 PM

Screen Shot 2022-05-12 at 4 19 28 PM

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added for moved, renamed, or deleted pages.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

**Comment by duplo83** [comment]
I love these changes! They are a major improvement, the sidebar is noticeably better and less cluttered. 

In regards to this change, "Removes the large header with the single "Terraform Enterprise" Deployment and Operation", would it be possible to still include 'Terraform' before Enterprise in the new menu link? If it spills over and create a link that is 2 lines in length, maybe its not worth it. I just fear that customers may look at the overall page and see that it is relevant to both Terraform Cloud and Terraform Enterprise and then see this link which is JUST 'Enterprise' and confuse it for some other enterprise consideration that is somehow distinct from 'Terraform Enterprise.' The fact that we don't typically use the term 'Enterprise' without 'Terraform' preceding it might confuse people. Maybe I'm overthinking it, but it stood out to me.

**Comment by laurapacilio** [comment]
@duplo83 - Really good call out on the naming. we're not actually supposed to separate the Terraform from the name..and I thought it might be okay in this case but...it's not. Thank you for flagging! I adjusted things so that we can say "Terraform Enterprise" without going to two lines. I actually think we don't even need "Install" --> Do you think that's implied with the administration part? LMK what you think!

File: content/cloud-docs/overview.mdx
Comment (judithpatudith) [inline content/cloud-docs/overview.mdx:50]:
```suggestion
[Organization owners](/cloud-docs/users-teams-organizations/teams#the-owners-team) can manage an organization's billing plan. The plan and billing settings include an integrated storefront, and you can subscribe to paid plans with a credit card.
```

Comment (judithpatudith) [reply content/cloud-docs/overview.mdx:50]: The business plan still requires that you talk to sales.

**Review by judithpatudith — APPROVED** [review]
This looks great!! I had a couple tiny nits but it's overall pretty great! approved!

**Review by korinne — APPROVED** [review]
These look like great changes to me, and the only feedback I have (which I've already relayed to Laura but just putting here for historical context) is to change the nav back from "Migrating existing resources" to "Migrating to Terraform Cloud". This is primarily because users might be thinking of `terraform import` when searching for migrating existing resources, rather than migrating an existing state to TFC. The older title also would be more useful for SEO. Other than that, fantastic job!

**Comment by laurapacilio** [comment]
@korinne thank you! Please let me know if there are others in TFC who should review this before I merge. Also! As we discussed, I put the old title back for "Migrating to Terraform Cloud". That was great feedback and I appreciate you catching that!

**Pull request merged**
